backend/src/main/kotlin/dev/kviklet/kviklet/proxy/postgres/Connection.kt

```diff
@@ -0,0 +1,99 @@
+package dev.kviklet.kviklet.proxy.postgres
+
+import dev.kviklet.kviklet.db.ExecutePayload
+import dev.kviklet.kviklet.service.EventService
+import dev.kviklet.kviklet.service.dto.ExecutionRequest
+import java.io.InputStream
+import java.io.OutputStream
+import java.net.Socket
+import java.net.SocketTimeoutException
+import java.nio.ByteBuffer
+
+
+class Connection(
+    clientSocket: Socket,
+    targetSocket: Socket,
+    private val eventService: EventService,
+    private val executionRequest: ExecutionRequest,
+    private val userId: String,
+) {
+    private var clientInput: InputStream = clientSocket.getInputStream()
+    private var clientOutput: OutputStream = clientSocket.getOutputStream()
+    private var serverInput: InputStream = targetSocket.getInputStream()
+    private var serverOutput: OutputStream = targetSocket.getOutputStream()
+    private val boundStatements: MutableMap<String, Statement> = mutableMapOf()
+    private var terminationMessageReceived: Boolean = false
+
+    fun startHandling() {
+        // This basically transfers messages between the client and the server sockets.
+        // NOTE: At this point the client connection is set up. SSL, Auth etc... are handled in ClientConnectionSetup.kt
+        while (!terminationMessageReceived) {
+            readDataFromSSLSocketIfAvailable(clientInput) { handleClientData(it) }
```

Comment:
This is actually not always an SSLSocket right? maybe just call it `readDataFromStream`.